### DISCUSSION TITLE
What is the best practise for "re-firing" a usePrepareContractWrite hook?

### DISCUSSION THREAD

**Discussion by hardingjam** [wevm/wagmi#2757] [Q&A]
I have a button component for purchasing tokens using a 'custom' ERC20 via the Zora Asks module. There are three steps to the purchase flow:

1. Approve the Zora module
2. Approve the ERC20 spend
3. Fill the Ask

I'm using the hook like so:

```

    const [currencyAddress, setCurrencyAddress] = useState<string>(
        ERC20Address
    );

  const {
        config: fillAskConfig,
        isSuccess: fillAskPrepSuccess,
        error: fillAskPrepError,
    } = usePrepareContractWrite({
        address: asksV1Address as `0x${string}`,
        abi: asksV1Abi,
        functionName: "fillAsk",
        chainId: 1
        args: [
            currentToken.address,
            BigNumber.from(currentToken.tokenId),
            currencyAddress,
            BigNumber.from((readData as any)?.askPrice?.toString() || "0"),
            constants.AddressZero,
        ],
        onSuccess(data) { 
          // This means that approval checks have passed
        },
             onError(data) {

           // This means that either the module approval or spend approval steps have not been completed

        },
}
   ```

I understand that the usePrepareContractWrite hook will refresh if the args change, so currently I am trying to force this by changing the currencyAddress in state (twice) after each successful approval transaction

```
          setCurrencyAddress("");
          setCurrencyAddress(ERC20Address);
```

This feels hacky and I wonder if there's an explicit way to re-prep the FillAsk function, or otherwise a recommended way to go about this.

**Comment by kasparkallas** [comment]
Use `refetch` which is returned by `usePrepareContractWrite`?
